oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -698,6 +704,21 @@ private void validatePasswordGrantType() {
 			Assert.hasText(this.tokenUri, "tokenUri cannot be empty");
 		}
 
+		private void validateAuthorizationGrantTypes() {
+			List<AuthorizationGrantType> authorizationGrantTypes =
+					Arrays.asList(AuthorizationGrantType.AUTHORIZATION_CODE, AuthorizationGrantType.CLIENT_CREDENTIALS,
+							AuthorizationGrantType.REFRESH_TOKEN, AuthorizationGrantType.IMPLICIT,
+							AuthorizationGrantType.PASSWORD);
+			for (AuthorizationGrantType authorizationGrantType : authorizationGrantTypes) {
+				if (authorizationGrantType.getValue().equalsIgnoreCase(this.authorizationGrantType.getValue()) &&
+						authorizationGrantType.equals(this.authorizationGrantType)) {
```

Comment:
Is this logic correct? should it be not-equals?

Reply:
nope, will fix